Template/Trigger/AllElementsClickTrigger.web.js

```diff
@@ -13,6 +13,11 @@
                 return;
             }
             var target = event.target;
+
+            if (target.shadowRoot) {
+                target = event.composedPath()[0];
```

Comment:
@multikoop Can we update it to below code and add a comment so that in future we remember why we need to select the first element for the event array
```suggestion
                var composedPath = event.composedPath();
                if (composedPath.length) {
                      target = composedPath[0];   //In shadow DOM keep select the first event path as the target
                }
```

Reply:
Thanks @AltamashShaikh  I have added the code change to all All*ClickTrigger.web.js.